Desktop/ruby_test/calender.rb

```diff
@@ -0,0 +1,71 @@
+require 'date'
+require 'optparse'
+
+opt = OptionParser.new
+opt.on('-m') { |v| }
+opt.parse!(ARGV)
+num = ARGV.to_a
+
+# 何も入力されていなければ今日の日付代入
+if num.empty?
+  time = Time.now
+  num[0] = time.month
+end
+
+# 例外処理
+begin
+  a = Date.new(2022, num[0].to_i, 1)
+rescue StandardError
+  puts "#{num[0]} is neither a month number (1..12) nor a name"
+  exit
+end
```

Comment:
例外処理よりも`num[0]`が1 ~ 12以外の場合にexitをするほうがよいです。
```suggestion
unless (1..12).include?(num[0]) 
  puts "#{num[0]} is neither a month number (1..12) nor a name"
  exit 
end
```

引数の値確認は可能な限りはじめのほうにやったほうがよいです。